### PR TITLE
A tasty assortment of phlegmatic niceties

### PR DESCRIPTION
**Pull request by livnev**

### CODE REVIEW

File: libexec/klab-prove [hunk 1]
```diff
@@ -48,7 +48,8 @@ K_OPTS=-Xmx10G \
       --no-alpha-renaming \
       --no-sort-collections \
       --output json \
-      --smt_prelude "$KLAB_OUT/prelude.smt2" \
+      --boundary-cells k,pc \
```
Comment (livnev) [inline libexec/klab-prove]: Don't we need to treat internal `lemma`s differently to normal `behaviour`s here?

Comment (livnev) [reply libexec/klab-prove]: @MrChico ^

File: libexec/klab-prove [hunk 2]
```diff
@@ -56,13 +56,13 @@ result=$?
 
 if [ "$result" == "0" ]; then
   echo "${green}Proof ${bold}ACCEPT${reset}:" "$(basename "$target_spec")" "$dump_notice"
-  mkdir -p $KLAB_OUT/pass && touch "$KLAB_OUT/pass/$spec_hash"
+  mkdir -p "$KLAB_OUT/pass" && touch "$KLAB_OUT/pass/$spec_hash"
 else
   if [ "$result" == "130" ]; then
-    echo 'catched'
+    echo 'Caught ^C'
```
Comment (MrChico) [inline libexec/klab-prove]: hmm

File: libexec/klab-prove (hunk 1 above)
Comment (MrChico) [reply libexec/klab-prove]: nope. The semantics here is that if both the RHS k and pc cells are matched by the current term, but the current term does not imply the target, we stop the execution. For internal specs, the k cell is matched by a lot of stuff, but not pc. For normal `behaviour`s the `pc` cell is matched by a lot of stuff, but not `k`.

File: libexec/klab-experience [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+klab prove-all "$@"
```
Comment (MrChico) [inline libexec/klab-experience]: 🙏

File: resources/rules.k.tmp [hunk 4]
```diff
@@ -172,13 +172,18 @@ rule ACCTCODE in SetItem( 1 )
...
+    rule #sizeWordStack(#take(N, _), M) => N +Int M
+
     syntax Bool ::= #noOverflow    ( WordStack ) [function]
                   | #noOverflowAux ( WordStack ) [function]
  // -------------------------------------------------------
-    rule #noOverflow(WS) => #sizeWordStack(WS) <=Int 32 andBool #noOverflowAux(WS)
-
-    rule #noOverflowAux(W : WS)     => 0 <=Int W andBool W <Int 256 andBool #noOverflowAux(WS)
-    rule #noOverflowAux(.WordStack) => true
+    rule #noOverflow(WS) => #sizeWordStack(WS) <=Int 32
```
Comment (MrChico) [inline resources/rules.k.tmp]: An awful lot of code for this macro

**Comment by MrChico** [comment]
Removing the STRING and BYTES imports removes that annoying "ambiguous parsing" warning. As far as I can tell, the only place where we used those imports was for `k-dss` filing functions, but the functionality there is better defined using functions already present in `data.md`: https://github.com/dapphub/k-dss/commit/e1115035a361bf926a9521f0af8238da2c59add3

File: libexec/klab-prove (hunk 1 above)
Comment (livnev) [reply libexec/klab-prove]: @MrChico got it.

**Comment by livnev** [comment]
> Removing the STRING and BYTES imports removes that annoying "ambiguous parsing" warning. As far as I can tell, the only place where we used those imports was for `k-dss` filing functions, but the functionality there is better defined using functions already present in `data.md`: [dapphub/k-dss@e111503](https://github.com/dapphub/k-dss/commit/e1115035a361bf926a9521f0af8238da2c59add3)

Great news that you figured out a way to get rid of those imports.

**Pull request merged**